src/python/pants/option/global_options.py

```diff
@@ -475,860 +486,764 @@ def from_options(cls, options: OptionValueContainer) -> LocalStoreOptions:
-                "There is at most one pantsd process per workspace."
-            ),
-        )
-
-        # These facilitate configuring the native engine.
-        register(
-            "--print-stacktrace",
-            advanced=True,
-            type=bool,
-            default=False,
-            help="Print the full exception stack trace for any errors.",
-        )
-        register(
-            "--engine-visualize-to",
-            advanced=True,
-            default=None,
-            type=dir_option,
-            help=(
-                "A directory to write execution and rule graphs to as `dot` files. The contents "
-                "of the directory will be overwritten if any filenames collide."
-            ),
-        )
-
-        # Pants Daemon options.
-        register(
-            "--pantsd-pailgun-port",
-            advanced=True,
-            type=int,
-            default=0,
-            daemon=True,
-            help="The port to bind the Pants nailgun server to. Defaults to a random port.",
-        )
-        register(
-            "--pantsd-invalidation-globs",
-            advanced=True,
-            type=list,
-            default=[],
-            daemon=True,
-            help="Filesystem events matching any of these globs will trigger a daemon restart. "
-            "Pants's own code, plugins, and `--pants-config-files` are inherently invalidated.",
-        )
+class BootstrapOptions:
+    """This are the set of options necessary to create a Scheduler. If an option is not consumed
+    during creation of a Scheduler, it should be in `register_options` instead.
 
-        process_execution_local_parallelism = "--process-execution-local-parallelism"
-        rule_threads_core = "--rule-threads-core"
-        rule_threads_max = "--rule-threads-max"
-
-        register(
-            rule_threads_core,
-            type=int,
-            default=max(2, CPU_COUNT // 2),
-            default_help_repr="max(2, #cores/2)",
-            advanced=True,
-            help=(
-                "The number of threads to keep active and ready to execute `@rule` logic (see "
-                f"also: `{rule_threads_max}`).\n\nValues less than 2 are not currently supported. "
-                "\n\nThis value is independent of the number of processes that may be spawned in "
-                f"parallel locally (controlled by `{process_execution_local_parallelism}`)."
-            ),
-        )
-        register(
-            rule_threads_max,
-            type=int,
-            default=None,
-            advanced=True,
-            help=(
-                "The maximum number of threads to use to execute `@rule` logic. Defaults to "
-                f"a small multiple of `{rule_threads_core}`."
-            ),
-        )
+    Bootstrap option values can be interpolated into the config file, and can be referenced
+    programmatically in registration code, e.g., as register.bootstrap.pants_workdir.
```

Comment:
I think this isn't possible with the new Options API right? Hence your `bin_name()` contribution. If we want to port other options like `--pants-distdir`, then we will need to implement a similar solution.

Reply:
It actually is possible, but since no one will have `register` anymore, it isn't important.

Reply:
Yeah, de facto it's not possible with the new API. I recommend deleting this.